SDL/SDLJoystick.cpp

```diff
@@ -41,16 +41,19 @@ SDLJoystick::SDLJoystick(bool init_SDL ) : registeredAsEventHandler(false) {
 		cout << "gamecontrollerdb.txt missing" << endl;
 	}
 	cout << "SUCCESS!" << endl;
-	setUpControllers();
+	//setUpControllers();
+	_njoy = njoy;
+
 }
 
 void SDLJoystick::setUpControllers() {
 	int numjoys = SDL_NumJoysticks();
-	for (int i = 0; i < numjoys; i++) {
-		setUpController(i);
-	}
-	if (controllers.size() > 0) {
-		cout << "pad 1 has been assigned to control pad: " << SDL_GameControllerName(controllers.front()) << endl;
+
+	if(_njoy < numjoys) {
+          setUpController(_njoy);
+	  if (controllers.size() > 0) {
+		cout << "pad 1 has been assigned to control pad: " << SDL_JoystickNameForIndex(_njoy) << endl;
```

Comment:
Indentation is all over the place here. Please use an editor that respects editorconfig, or just manually switch to tabs, 4-width, when editing PPSSPP.